tmt/base.py

```diff
@@ -1529,6 +1529,45 @@ def lint_require_type_field(self) -> LinterReturn:
         yield LinterOutcome.FIXED, 'added type to requirements'
 
 
+@dataclasses.dataclass(repr=False)
+class Tests(tmt.lint.Lintable['Tests']):
```

Comment:
I would make it very clear that this `Tests` class is not representing some set of test for any other purpose than lining them as a collection of objects. We have `Test` class representing a test, and `Tests` suggests it has more to do with test collections than just the lining.

Since plans and stories also have IDs, maybe we should have a generic collection to hold `tmt.base.Core` instances, no matter what exactly they are, and lint them. `tmt test lint` would fill it with `Test` instances, `tmt lint` would fill it with tests, plans and stories - I suppose test ID should not conflict with story ID either.

Reply:
> > Since plans and stories also have IDs, maybe we should have a generic collection to hold tmt.base.Core instances
> 
> Good point! I changed it to hold either a list of `tmt.base.Test`, `tmt.base.Plan` or `tmt.base.Story`. And since naming is always the most difficult part: I am not sure I found a good name with `LintableCollection`. What do you think?

`LintableCollection` sounds good.

WRT test, plan or story, can you try it out with `list[Core]` instead of union of homogenous lists of subclasses? All three classes are derived from `Core`, `Core` defines `id`, it should work.

> > I suppose test ID should not conflict with story ID either.
> 
> Ah, didn't know about this. With the current approach I can define only lint rules for collections of one type of object like Tests or Stories.

Switching to `Core` might help.

> Thinking about using it: Should `tmt test lint` then inspect all `tmt.base.Test` and `tmt.base.Story`, and show an error if a duplicate id is found? In that case it wouldn't lint only tests.

I'd stick to what `tmt * lint` inspects: `tmt test|plan|story lint` fills the collection with all kinds of objects, and I would get them validated all against each other, i.e. I'd be able to see a test conflicting with a story. `tmt test lint` would fill it with tests only, therefore no validation between kinds of tmt objects.

But this is a subjective opinion, I can also imagine this cross-examination might be interesting enough that it would be enabled in general, for all subcommand, and `tmt test lint` would run test linters for tests and collection linters for everything too. Let's see what others think.

Reply:
> I'd stick to what tmt * lint inspects: tmt test|plan|story lint fills the collection with all kinds of objects, and I would get them validated all against each other, i.e. I'd be able to see a test conflicting with a story. tmt test lint would fill it with tests only, therefore no validation between kinds of tmt objects.

That makes sense - and also from a user point of view might be quite helpful to identify duplicate ids across multple types.  

> WRT test, plan or story, can you try it out with list[Core] instead of union of homogenous lists of subclasses? All three classes are derived from Core, Core defines id, it should work.

Yes, you are right. Switching to `Core` required a bit of refactoring, but now it is also quite easy to extend with further lint rules on such sets.